Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -119,6 +124,100 @@ try {
         exit
     }
 
+    $buildArtifactFolder = Join-Path $projectPath ".buildartifacts"
+    New-Item $buildArtifactFolder -ItemType Directory | Out-Null
+
+    if ($baselineWorkflowSHA -and $baselineWorkflowRunId -ne '0' -and $settings.incrementalBuilds.mode -eq 'modifiedApps') {
+        # Incremental builds are enabled and we are only building modified apps
+        $headSHA = git rev-parse HEAD
+        Write-Host "Current HEAD is $headSHA"
+        git fetch origin $baselineWorkflowSHA | Out-Host
+        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to fetch baseline SHA $baselineSHA" }
+        Push-Location $ENV:GITHUB_WORKSPACE
+        Write-Host "git diff --name-only $baselineWorkflowSHA $headSHA"
+        $modifiedFiles = @(git diff --name-only $baselineWorkflowSHA $headSHA | ForEach-Object { "$_".Replace('/', [System.IO.Path]::DirectorySeparatorChar) })
+        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to diff baseline SHA $baselineSHA with current HEAD $headSHA" }
+        Pop-Location
+        OutputMessageAndArray -message "Modified files" -arrayOfStrings $modifiedFiles
+        $buildAll = Get-BuildAllApps -baseFolder $baseFolder -project $project -modifiedFiles $modifiedFiles
+        if (!$buildAll) {
+            $skipFolders = @()
+            $unknownDependencies = @()
+            $knownApps = @()
+            $allFolders = @(GetFoldersFromAllProjects -baseFolder $baseFolder | ForEach-Object { $_.Replace('\', $([System.IO.Path]::DirectorySeparatorChar)).Replace('/', $([System.IO.Path]::DirectorySeparatorChar)) } )
+            Push-Location $ENV:GITHUB_WORKSPACE
+            $modifiedFolders = @($allFolders | Where-Object {
+                $modifiedFiles -like "$($_)$([System.IO.Path]::DirectorySeparatorChar)*"
+            })
+            Pop-Location
+            OutputMessageAndArray -message "Modified folders" -arrayOfStrings $modifiedFolders
+            Sort-AppFoldersByDependencies -appFolders $allFolders -baseFolder $baseFolder -skippedApps ([ref] $skipFolders) -unknownDependencies ([ref]$unknownDependencies) -knownApps ([ref] $knownApps) -selectSubordinates $modifiedFolders | Out-Null
+            OutputMessageAndArray -message "Skip folders" -arrayOfStrings $skipFolders
+            $downloadAppFolders = @($settings.appFolders | Where-Object { Write-Host "check '$project$([System.IO.Path]::DirectorySeparatorChar)$($_.SubString(2))'"; $skipFolders -contains "$project$([System.IO.Path]::DirectorySeparatorChar)$($_.SubString(2))" })
```

Comment:
Why Substring(2)? 🤔

Reply:
I think it's because `appFolders` are relative to the project and always start with "./".

But I agree it isn't very easily understandable.